### ISSUE TITLE
More missing derivations in VLog

### ISSUE THREAD

**Issue by mkroetzsch** [karmaresearch/vlog#59]
This might be related to #58, but it could also have another reason. To reproduce, use the following input to VLog:
```
element(L,E) :- TRIPLE(L,first,E)
next(L1,L2) :- TRIPLE(L1,rest,L2),element(L2,E)
Intersection(Y) :- TRIPLE(X,intersectionOf,Y)
hasList(X,Y)   :- TRIPLE(X,intersectionOf,Y)
Intersection(Z) :- Intersection(Y),next(Y,Z)
hasList(X,Z) :- hasList(X,Y),next(Y,Z)
hasIntersectionElement(X,E) :- hasList(X,Y),Intersection(Y),element(Y,E)
```
EDB configuration edb.conf:
```
EDB0_predname=TRIPLE
EDB0_type=INMEMORY
EDB0_param0=.
EDB0_param1=triples
```
Data file triples.csv contains:
```
B-fa5724d6-ca7b-30d2-8444-297a0b98ebc3,intersectionOf,B-360d4bb5-c9d3-3c12-8f6d-59e7651598e2
B-360d4bb5-c9d3-3c12-8f6d-59e7651598e2,first,Woman
B-360d4bb5-c9d3-3c12-8f6d-59e7651598e2,rest,B-1a790bdd-6a06-3bfc-8567-7703fdad45f6
B-1a790bdd-6a06-3bfc-8567-7703fdad45f6,first,Parent
B-1a790bdd-6a06-3bfc-8567-7703fdad45f6,rest,nil
```

The materialisation contains:
- element
```
2       B-360d4bb5-c9d3-3c12-8f6d-59e7651598e2  Woman
2       B-1a790bdd-6a06-3bfc-8567-7703fdad45f6  Parent
```
-  Interrsection
```
4       B-360d4bb5-c9d3-3c12-8f6d-59e7651598e2
9       B-1a790bdd-6a06-3bfc-8567-7703fdad45f6
```
- hasList
```
5       B-fa5724d6-ca7b-30d2-8444-297a0b98ebc3  B-360d4bb5-c9d3-3c12-8f6d-59e7651598e2
11      B-fa5724d6-ca7b-30d2-8444-297a0b98ebc3  B-1a790bdd-6a06-3bfc-8567-7703fdad45f6
```
- hasIntersectionElement
```
12      B-fa5724d6-ca7b-30d2-8444-297a0b98ebc3  Parent
```

However, the rule ```hasIntersectionElement(X,E) :- hasList(X,Y),Intersection(Y),element(Y,E)``` should have two conclusions in this situation.

**Comment by CerielJacobs** [comment]
Fixed.

**Comment by CerielJacobs** [comment]
Note: this also fixes issue #58
